aten/src/ATen/TensorIterator.cpp

```diff
@@ -940,10 +940,6 @@ void TensorIteratorBase::compute_mem_overlaps(const TensorIteratorConfig& config
   if (!config.check_mem_overlap_) {
     return;
   }
-  if (is_meta_) {
```

Comment:
Just curious, do you want to take this out for more semantic reasons or perf reasons?

Semantic: We no longer have to special case meta tensors for memory overlap, it "just works".

perf: We get one-less branch in happy-path TI code, although the benefit is probably minimal. Conversely, if we start to care about meta tensor performance we might want to add this back in.

Reply:
It's semantic reasons!